api/src/term/matcher/_graph_name_matcher.rs

```diff
@@ -135,3 +135,19 @@ impl GraphNameMatcher for Any {
         true
     }
 }
+
+impl<M: GraphNameMatcher> GraphNameMatcher for Not<M> {
+    type Term = SimpleTerm<'static>; // not actually used
+
+    fn matches<T2: Term + ?Sized>(&self, graph_name: GraphName<&T2>) -> bool {
+        !self.0.matches(graph_name)
+    }
+}
+
+impl GraphNameMatcher for Option<SimpleTerm<'static>> {
+    type Term = SimpleTerm<'static>;
+
+    fn matches<T2: Term + ?Sized>(&self, graph_name: GraphName<&T2>) -> bool {
+        graph_name_eq(self.as_ref(), graph_name.map(Term::as_simple))
+    }
+}
```

Comment:
I'm not comfortable with this, as Option<T> as (Graph)TermMatcher currently are used to mean "match the value of the option if any, otherwise match *nothing*". This impl would create confusion.

Note that, for any term `t` (resp. graphname `g`), wrapping it in sqare-brackets makes a matcher, so you by writing `Not([Some(my_graph_name)])` you would achieve exactly what you want, without the need to add this new impl.